### DISCUSSION TITLE
Adding GLMsingle to Neurodesk?

### DISCUSSION THREAD

**Discussion by jacob-prince** [neurodesk/neurodesk.github.io-hugo#416] [New Tool]
Hi, we have recently created [GLMsingle](https://github.com/cvnlab/GLMsingle), a pipeline for improving the accuracy of GLM estimation in fMRI, available in Matlab and Python. GLMsingle is very easy to install from command line. When run, it automatically implements a set of data-driven optimizations to help denoise signal estimates, and is particularly useful for single-trial/condition-rich design matrices. Our pipeline was initially used for response estimation in the popular NSD dataset, and is being increasingly adopted by users in cognitive neuroscience / neuroAI. We feel it could be useful to incorporate into NeuroDesk.

**If there would be collective interest in incorporating GLMsingle into NeuroDesk, please advise as to how best to proceed?** 

Resources: 
GLMsingle GitHub repo with example scripts: https://github.com/cvnlab/GLMsingle
GLMsingle manuscript: https://elifesciences.org/articles/77599
NSD dataset: https://naturalscenesdataset.org/
NSD manuscript: https://www.nature.com/articles/s41593-021-00962-x

**Comment by stebo85** [comment]
Dear @jacob-prince,

It would be great to have GLMsingle in Neurodesk :) Would you like to propose an addition via a pull request to https://github.com/NeuroDesk/neurocontainers ?

Here is a template for a new tool https://github.com/NeuroDesk/neurocontainers/tree/master/recipes/template

Let us know if you need any help

Thank you
Steffen

**Comment by jacob-prince** [comment]
Thanks for the quick response! We will work on a pull request and follow up as necessary if we run into any issues.

Best,
Jacob

**Comment by sjp117** [comment]
Hi, I'm helping to add GLMsingle to neurocontainer.
Our matlab component involves adding paths to be used in matlab.
https://github.com/cvnlab/GLMsingle/blob/a232c1945287943f8e1e009cef5e7e915d24fb1e/setup.m

What would be the best way to implement this?
I'm thinking that the our container would load the matlab module and set a startup.m script to add the needed paths.
But what do you advise?

**Reply by stebo85** [reply]
That should work! However, to increase the usefulness of the container, would it be possible to compile your matlab components so they work with the matlab runtime environment?

**Reply by sjp117** [reply]
Hi @stebo85,
Thank you for the response!
We will stick with our original intention since the matlab component is intended to be used like our python package.
However, I am unsure how to design the build.sh recipe for neurocontainer, for our specific case. How should I go about having it load the matlab module? And what directory should the startup.m script be placed? (usually they would go to the ~/.matlab of the user home directory)

**Reply by stebo85** [reply]
Dear @sjp117,

Ok, if you don't want to provide a compiled version of your matlab toolboox, then it doesn't need a new container. If you simply want to run your matlab script in Neurodesk, you can do this already by just cloning your repository and then executing it in matlab - exactly as you describe in your github repository. 

Is the Python version equivalent in terms of features? Then this would be a better candidate for building a container and making that available in Neurodesk.


Thank you
Steffen

**Reply by stebo85** [reply]
Dear @sjp117 ,

Thank you for the contribution. This is now integrated.

Thank you
Steffen
